src/libcore/ops/drop.rs

```diff
@@ -1,85 +1,120 @@
-///
-/// # Examples
-///
+/// Destructors for cleaning up resources
+/// 
+/// When a value is no longer needed, Rust will run a "destuctor" on that value.
+/// The most common way that a value is no longer needed is when it goes out of
+/// scope. Destuctors may still run in other circumstances, but we're going to
+/// focus on scope for the examples here. To learn about some of those other cases,
+/// please see [the reference] section on destructors.
+/// 
+/// [the reference]: https://doc.rust-lang.org/reference/destructors.html
+/// 
+/// The `Drop` trait provides a way to implement a custom destructor on a type.
+/// Why would you want such a thing? Well, many types aren't only data: they
+/// manage some sort of resource. That resource may be memory, it may be a file
+/// descriptor, it may be a network socket. But when the type is no longer going
+/// to be used, it should "clean up" that resource by freeing the memory or
+/// closing the file or socket. This is the job of a desctructor, and therefore
+/// the job for `Drop::drop`.
+/// 
 /// ## Implementing `Drop`
-///
-/// The `drop` method is called when `_x` goes out of scope, and therefore
-/// `main` prints `Dropping!`.
-///
-/// ```
+/// 
+/// When a value goes out of scope, it will call `Drop::drop` on that value. For example,
+/// 
+/// ```rust
 /// struct HasDrop;
-///
+/// 
 /// impl Drop for HasDrop {
 ///     fn drop(&mut self) {
 ///         println!("Dropping!");
 ///     }
 /// }
-///
+/// 
 /// fn main() {
 ///     let _x = HasDrop;
 /// }
 /// ```
-///
-/// ## Dropping is done recursively
-///
-/// When `outer` goes out of scope, the `drop` method will be called first for
-/// `Outer`, then for `Inner`. Therefore, `main` prints `Dropping Outer!` and
-/// then `Dropping Inner!`.
-///
-/// ```
-/// struct Inner;
-/// struct Outer(Inner);
-///
-/// impl Drop for Inner {
-///     fn drop(&mut self) {
-///         println!("Dropping Inner!");
-///     }
+/// 
+/// Here, `main` will print `Dropping!`. In other words, the compiler generates code that
+/// kind of looks like this:
+/// 
+/// ```rust,ignore
+/// fn main() {
+///     let mut _x = HasDrop;
+/// 
+///     Drop::drop(&mut x);
 /// }
-///
-/// impl Drop for Outer {
+/// ```
+/// 
+/// As you can see, our custom implementation of `Drop` will be called at the end of `main`.
+/// 
+/// ## You cannot call `Drop::drop` yourself
+/// 
+/// Because the compiler automatically calls `Drop::drop`, you cannot call it yourself. This
+/// would lead to "double drop", where `Drop::drop` is called twice on the same value. This
+/// can lead to things like "double frees".
+/// 
+/// In other words, if you tried to write the code in the previous example with the explicit
+/// call to `Drop::drop`, you'd get a compiler error.
+/// 
+/// If you'd like to call `drop` yourself, there is something you can do: call [`std::mem::drop`].
+/// This function will drop its argument. For more, see its documentation.
```

Comment:
> For more, see its documentation.

Is this required when the function itself is linked?

Reply:
I'm trying to allude to there being more to it, if that makes any sense.